src/report_coverage.js

```diff
@@ -0,0 +1,55 @@
+const path  = require('path')
+
+function report_coverage(json_file, report_stream, regex_pattern, fail_threshold) {
+    report_stream.write(`${'Name'.padEnd(55)} ${'Stmts'.padStart(6)} ${'Miss'.padStart(6)} `)
+    report_stream.write(`${'Branch'.padStart(6)} ${'BrPart'.padStart(6)} ${'Cover'.padStart(6)}\n`)
+    report_stream.write('\n'.padStart(91, '-'))
+
+    const regex = new RegExp(regex_pattern)
+    let statements_total = 0, statements_covered = 0, branches_total = 0, branches_covered = 0
+
+    for (const report_property in json_file) {
+        if (report_property === 'total')
+            continue
+        const js_path = path.relative(__dirname, report_property)
+        if (!regex.test(report_property))
+            continue
+
+        const json_data = json_file[report_property]
+
+        report_stream.write(js_path.padEnd(55))
+        report_stream.write(json_data.statements.total.toString().padStart(7))
+        report_stream.write((json_data.statements.total - json_data.statements.covered).toString().padStart(7))
+        report_stream.write(json_data.branches.total.toString().padStart(7))
+        report_stream.write((json_data.branches.total - json_data.branches.skipped).toString().padStart(7))
+        report_stream.write(`${Math.round(((json_data.statements.covered + json_data.branches.covered)
+            / (json_data.statements.total + json_data.branches.total)) * 100)}%`.padStart(7))
+        report_stream.write('\n')
+
+        statements_total += json_data.statements.total
+        statements_covered += json_data.statements.covered
+        branches_total += json_data.branches.total
+        branches_covered += json_data.branches.covered
+    }
+
+    if (statements_total + branches_total === 0) {
+        console.err('There are no matching files or all matching files are empty.')
+        process.exit(1)
```

Comment:
We should also cover this case in tests.

Reply:
This is covered now.